src/service/authService.ts

```diff
@@ -0,0 +1,54 @@
+import sendRequest from "@/lib/sendRequest";
+import { LoginResponse } from "@/models/AuthModels/loginResponse";
+import { RegisterResponse } from "@/models/AuthModels/registerResponse";
+
+//todo: add interfaces for the response
+export default {
+
+  /**
+   * register a new user
+   * @param username
+   * @param password 
+   * @returns a promise that resolves to the response from the server
+   */
+  register: async (username: string, password: string) => {
+    return await sendRequest({
+      method: 'POST',
+      endpoint: '/auth/register',
+      body: {
+        username,
+        password
+      }
+    }) as Promise<RegisterResponse>;
```

Comment:
for this can you just define return type next to function parameters or do you have to do it like this